revocation/crl/fetcher.go

```diff
@@ -117,19 +126,110 @@ func (f *HTTPFetcher) fetch(ctx context.Context, url string) (*Bundle, error) {
 		return nil, err
 	}
 
-	// check delta CRL
-	// TODO: support delta CRL https://github.com/notaryproject/notation-core-go/issues/228
-	for _, ext := range base.Extensions {
-		if ext.Id.Equal(oidFreshestCRL) {
-			return nil, errors.New("delta CRL is not supported")
-		}
+	// fetch delta CRL from base CRL extension
+	deltaCRL, err := f.fetchDeltaCRL(ctx, base.Extensions)
+	if err != nil && !errors.Is(err, errDeltaCRLNotFound) {
+		return nil, err
 	}
 
 	return &Bundle{
-		BaseCRL: base,
+		BaseCRL:  base,
+		DeltaCRL: deltaCRL,
 	}, nil
 }
 
+// fetchDeltaCRL fetches the delta CRL from the given extensions of base CRL.
+//
+// It returns errDeltaCRLNotFound if the delta CRL is not found.
+func (f *HTTPFetcher) fetchDeltaCRL(ctx context.Context, extensions []pkix.Extension) (*x509.RevocationList, error) {
+	idx := slices.IndexFunc(extensions, func(ext pkix.Extension) bool {
+		return ext.Id.Equal(oidFreshestCRL)
+	})
+	if idx == -1 {
```

Comment:
Just to be safe, it is better to have
```suggestion
	if idx < 0 {
```

Reply:
Updated.